src/github.com/getlantern/balancer/strategy.go

```diff
@@ -97,9 +100,11 @@ func (s *dialerHeap) Pop() interface{} {
 	return x
 }
 
-func (s *dialerHeap) AuthTokens() (tokens []string) {
+func (s *dialerHeap) onRequest(req *http.Request) (tokens []string) {
 	for _, d := range s.dialers {
-		tokens = append(tokens, d.AuthToken)
+		if d.OnRequest != nil {
+			d.OnRequest(req)
+		}
```

Comment:
It also seems like theoretically we should have locking on s.dialers no? I guess Push and Pop aren't called much in practice, but if they were it seems like we'd have a bunch of race issues @oxtoacart (not specific to this PR other than the introduction of iterating in a pretty heavily used portion of the code)